baselines/fedavgm/fedavgm/strategy.py

```diff
@@ -0,0 +1,189 @@
+            Function used to configure validation. Defaults to None.
+        accept_failures : bool, optional
+            Whether or not accept rounds containing failures. Defaults to True.
+        initial_parameters : Parameters
+            Initial global model parameters.
+        server_learning_rate: float
+            Server-side learning rate used in server-side optimization.
+            Defaults to 1.0.
+        server_momentum: float
+            Server-side momentum factor used for FedAvgM. Defaults to 0.9.
+        """
+        super().__init__(
+            fraction_fit=fraction_fit,
+            fraction_evaluate=fraction_evaluate,
+            min_fit_clients=min_fit_clients,
+            min_evaluate_clients=min_evaluate_clients,
+            min_available_clients=min_available_clients,
+            evaluate_fn=evaluate_fn,
+            on_fit_config_fn=on_fit_config_fn,
+            on_evaluate_config_fn=on_evaluate_config_fn,
+            accept_failures=accept_failures,
+            initial_parameters=initial_parameters,
+            fit_metrics_aggregation_fn=fit_metrics_aggregation_fn,
+            evaluate_metrics_aggregation_fn=evaluate_metrics_aggregation_fn,
+        )
+        self.server_learning_rate = server_learning_rate
+        self.server_momentum = server_momentum
+        self.momentum_vector: Optional[NDArrays] = None
+
+    def __repr__(self) -> str:
+        """Compute a string representation of the strategy."""
+        rep = f"FedAvgM(accept_failures={self.accept_failures})"
+        return rep
+
+    def initialize_parameters(
+        self, client_manager: ClientManager
+    ) -> Optional[Parameters]:
+        """Initialize global model parameters."""
+        return self.initial_parameters
+
+    def aggregate_fit(
+        self,
+        server_round: int,
+        results: List[Tuple[ClientProxy, FitRes]],
+        failures: List[Union[Tuple[ClientProxy, FitRes], BaseException]],
+    ) -> Tuple[Optional[Parameters], Dict[str, Scalar]]:
+        """Aggregate fit results using weighted average."""
+        if not results:
+            return None, {}
+
+        # Do not aggregate if there are failures and failures are not accepted
+        if not self.accept_failures and failures:
+            return None, {}
+
+        # Convert results
+        weights_results = [
+            (parameters_to_ndarrays(fit_res.parameters), fit_res.num_examples)
+            for _, fit_res in results
+        ]
+
+        fedavg_result = aggregate(weights_results)  # parameters_aggregated from FedAvg
+
+        # original implementation follows convention described in
+        # https://pytorch.org/docs/stable/generated/torch.optim.SGD.html
+
+        # remember that updates are the opposite of gradients
+        pseudo_gradient: NDArrays = [
+            x - y
+            for x, y in zip(
+                parameters_to_ndarrays(self.initial_parameters), fedavg_result
+            )
+        ]
```

Comment:
The reason for your `mypy` error is because `self.initial_parameters` can be `None` if it hasn't been set. And a `None` is not expected by `parameters_to_ndarrays`. You can either put all that behind a `if` statement or put an `assert` at the top of your `aggregate_fit()` `initial_parameters` should always be set. For example:
```
        assert self.initial_parameters is not None, "initial parametrs must be set"
```

Reply:
This will also fix the second error you get

Reply:
Thank for the suggestion! I proceeded with that using the `assert`.